float64 pos_x
float64 pos_y
float64 pos_z
float64 post_x
float64 post_y
float64 post_z